manual-workflows/run_immuno_compute1.sh

```diff
@@ -0,0 +1,272 @@
+#!/bin/bash
+
+# ---------------------------------------------------------------------------------------------------------------------------------------------------------
+# Script to submit jobs for single or multiple samples to run the immuno pipeline on storage1 using bsub
+# Usage: bash run_immuno_compute1.sh --sample "Hu_254" --scratch_dir "/scratch1/fs1/mgriffit/jyao/miller_immuno" --job_group "/j.x.yao/2_job"
+# Usage for multiple samples: bash run_immuno_compute1.sh --sample "Hu_344 Hu_048" --scratch_dir "/scratch1/fs1/mgriffit/jyao/miller_immuno" --job_group "/j.x.yao/2_job"
+
+# Function to display usage
+usage() {
+    echo "Usage: $0 --sample <sample_id(s)> --scratch_dir <scratch_directory> --job_group <job_group>"
+    echo "  --sample: Single sample ID (e.g., 'Hu_254') or multiple sample IDs (e.g., 'Hu_344 Hu_048')"
+    echo "  --scratch_dir: Scratch directory to save pipeline artifacts"
+    echo "  --job_group: Your LSF job group"
+    echo ""
+    echo "Examples:"
+    echo "  $0 --sample 'Hu_254' --scratch_dir '/scratch1/fs1/mgriffit/jyao/miller_immuno' --job_group '/j.x.yao/2_job'"
+    echo "  $0 --sample 'Hu_344 Hu_048' --scratch_dir '/scratch1/fs1/mgriffit/jyao/miller_immuno' --job_group '/j.x.yao/2_job'"
+    exit 1
+}
+
+# Parse command line arguments
+while [[ $# -gt 0 ]]; do
+    case $1 in
+        --sample)
+            SAMPLES_STR="$2"
+            shift 2
+            ;;
+        --scratch_dir)
+            SCRATCH_DIR="$2"
+            shift 2
+            ;;
+        --job_group)
+            JOB_GROUP="$2"
+            shift 2
+            ;;
+        --help|-h)
+            usage
+            ;;
+        *)
+            echo "Unknown option: $1"
+            usage
+            ;;
+    esac
+done
+
+# Check if all required arguments are provided
+if [[ -z "$SAMPLES_STR" || -z "$SCRATCH_DIR" || -z "$JOB_GROUP" ]]; then
+    echo "Error: Missing required arguments"
+    usage
+fi
+
+# Convert sample string to array, work for a single sample or multiple samples
+SAMPLES=($SAMPLES_STR)
+
+# ----------------------------------------------------------------------------------------------------------------------------------------------------------
+# Get the directory where this script is located
+SCRIPT_DIR="$(cd "$(dirname "${BASH_SOURCE[0]}")" && pwd)"
+# Set WORK_DIR to two levels up from the script's directory
+WORK_DIR="$(cd "$SCRIPT_DIR/../.." && pwd)" # change this to absolute path
```

Comment:
This kind of thing leads to so much confusion in my opinion.  I would just make the work_dir another input to this script.  This has the added advantage that when people save records of their run_immuno_compute1.sh commands it is more obvious what is going on.